src/Azure.Functions.Sdk/FunctionsAssemblyScanner.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.IO.Abstractions;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+using Mono.Cecil;
+using ILogger = NuGet.Common.ILogger;
+
+namespace Azure.Functions.Sdk;
+
+public sealed partial class FunctionsAssemblyScanner
+{
+    private static readonly Regex ExcludedPackagesRegex = new(
+        @"^(System|Azure\.Core|Azure\.Identity|Microsoft\.Bcl|Microsoft\.Extensions|Microsoft\.Identity|Microsoft\.NETCore|Microsoft\.NETStandard|Microsoft\.Win32|Grpc)\..*",
```

Comment:
Do we want to use `GeneratedRegex` for better perf?

Reply:
Wish I could, but it is only available in net7+. This is netstandard2.0